client/files.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2022 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+import (
+	"io/fs"
+	"net/url"
+	"os"
+	"strconv"
+	"time"
+)
+
+// ListFilesOptions holds the options for a call to ListFiles
+type ListFilesOptions struct {
+	// Path is the absolute path of the fiole system entry to list. Note that
+	// when Pattern is specified, it should specify the absolute path of the
+	// parent directory (required)
+	Path string
+
+	// Pattern is the glob-like pattern string to filter results by. When
+	// present, only file system entries with names matching this pattern will
+	// be included in the call results.
+	Pattern string
+
+	// Itself, when set, will force directory entries not to be listed, but
+	// instead have their information returned as if they were regular files.
+	Itself bool
+}
+
+type FileInfo struct {
+	name    string
+	size    int64
+	mode    fs.FileMode
+	modTime time.Time
+
+	// Path is the full absolute path of the file
+	Path string
```

Comment:
Per other discussion, I think these should be methods too, for consistency (so not half of them are methods and half fields).